test/test_rides.py

```diff
@@ -0,0 +1,615 @@
+import unittest
+import json
+from app.views.views import app
+
+
+class viewsTest(unittest.TestCase):
+	def setUp(self):
+		"""
+		set up instances
+		"""
+		self.test = app.test_client(self)
+		self.headers = {'Content-type': 'application/json'}
+		self.post_ride = {
+		"username": "string",
+		"capacity": 10,
+		"destination": "string",
+		"month": "June",
+		"pickup": "sting",
+		"time": "string",
+		"year": 2018,
+		"date": "string",
+		"email": "string@gmail.com"
+		}
+
+
+		self.post_request = {
+		"username": "",
+		"pick_up_point": "string",
+		"destination": "string",
+		"pick_up_time": " s "
+		}
+
+	def tearDown(self):
+		"""
+		Realese flask app instances
+		"""
+		self.test = None
+		self.headers = None
+		self.post_ride = None
+
+
+	def test_no_ride(self):
+		"""
+		Test status code of get  rides
+		"""
+		response = self.test.post('/rides',
+			data=json.dumps(self.post_ride),headers=self.headers)
+		self.assertEqual(response.status_code,201)
+		response2 =self.test.get('/rides',headers=self.headers)
+		self.assertEqual(response2.status_code,200)
+
+
+
+	def test_single_data(self):
+		"""
+		test when ride account does not exist
+		"""
+		response = self.test.get('/rides/-34',headers=self.headers)
+		self.assertEqual(response.status_code,404)
+		self.assertIn('invalid ride id',response.data)
+
+	def test_single_dataFound(self):
+		"""
+		test when ride account exist
+		"""
+		response = self.test.post('/rides',
+			data=json.dumps(self.post_ride),headers=self.headers)
+		response2 = self.test.get('rides/0',headers=self.headers)
+		self.assertEqual(response2.status_code,200)
+		self.assertIn('username',response2.data)
+
+	def test_post_ride(self):
+		"""
+		test for when posting a ride
+		"""
+		response = self.test.post('/rides',
+			data=json.dumps(self.post_ride),headers=self.headers)
+		self.assertEqual(response.status_code,201)
+		self.assertIn('Content Uploaded',response.data)
+
+	def test_post_keys(self):
+		"""
+		test for when posting with the wrong keys
+		"""
+		post_ride = {
```

Comment:
create one `post_ride ` on setUp and re-use it rather than repeating yourself..

Reply:
I wanted to test different occurrence of post_ride eg post_ride where post_ride['month'] = 'not a month' or where post_ride['month'] = '      '